### PR TITLE
Wind rose

### PR DESCRIPTION
**Pull request by MRossol**
Wind Rose calculator w/ tests (take that...)

### CODE REVIEW

File: rex/utilities/utilities.py [hunk 1]
```diff
@@ -590,3 +590,144 @@ def roll_timeseries(arr, timezones):
...
+    ds_len : int
+        Length of dataset axis to chunk
+    chunk_size : int
+        Size of chunks
+
+    Returns
+    -------
+    chunks : list
+        List of chunk start and end positions
+        [(s_i, e_i), (s_i+1, e_i+1), ...]
```
Comment (grantbuster) [inline rex/utilities/utilities.py:610]: the title of this method would imply that this outputs a list of slices.

**Review by grantbuster — CHANGES_REQUESTED** [review]

File: rex/renewable_resource.py [hunk 2]
```diff
@@ -371,6 +371,35 @@ def __init__(self, h5_file, unscale=True, hsds=False, str_decode=True,
...
 
+    @property
+    def heights(self):
+        """
+        Extract available heights for pressure, temperature, windspeed, precip,
+        and winddirection variables. Used for interpolation/extrapolation.
+
+        Returns
+        -------
+        self._heights : list
```
[the remark this answers is not in the document]
Comment (MRossol) [reply rex/renewable_resource.py:382]: nope its a dict, fixing

File: rex/renewable_resource.py [hunk 3]
```diff
@@ -711,6 +711,57 @@ def circular_interp(cls, ts_1, h_1, ts_2, h_2, h):
 
         return out
 
+    def get_attrs(self, dset=None):
```
[the remark this answers is not in the document]
Comment (MRossol) [reply rex/renewable_resource.py:714]: but then we'd have to pre-load all of the attributes. It's not a ton of data, but also not sure it's necessary, thoughts?

File: rex/temporal_stats/wind_rose.py [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+        Resource class to use to access wind_h5
+
+        Returns
+        -------
+        Class
+        """
+        return self._res_cls
+
+    @staticmethod
+    def _compute_wind_rose(wspd, wdir, wspd_bins, wdir_bins):
```
[the remark this answers is not in the document]
Comment (MRossol) [reply rex/temporal_stats/wind_rose.py:66]: roger

File: rex/temporal_stats/wind_rose.py [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+            Time-series vector of complimentary wind direction
+        wspd_bins : ndarray
+            Wind speed bin edges
+        wdir_bins : ndarray
+            Wind direction bin edges
+
+        Returns
+        -------
+        wind_rose : ndarray
+            Vector of wind rose frequencies
```
[the remark this answers is not in the document]
Comment (MRossol) [reply rex/temporal_stats/wind_rose.py:85]: will do

File: rex/temporal_stats/wind_rose.py [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+            bin range stop value
+        step : int
+            bin range step value
+
+        Returns
+        -------
+        bin_edges : ndarray
+            Vector of inclusive bin edges
+        """
+        bin_edges = np.arange(start, stop + step, step)
```
[the remark this answers is not in the document]
Comment (MRossol) [reply rex/temporal_stats/wind_rose.py:112]: two reasons:
1) It's nice to be able to set the step size
2) I need to compute the bin edges for the DataFrame index. this approach seems more intuitive than n_bins + range

File: rex/temporal_stats/wind_rose.py [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+        else:
+            logger.info('Writing wind rose site meta data to {}'
+                        .format(out_fpath))
+            with self.res_cls(self.wind_h5) as f:
+                meta = f['meta', wind_rose.columns.values]
+
+            meta.to_csv(out_fpath, index=False)
+
+    @classmethod
+    def run(cls, wind_h5, hub_height, sites=None, wspd_bins=(0, 30, 1),
```
[the remark this answers is not in the document]
Comment (MRossol) [reply rex/temporal_stats/wind_rose.py:332]: how about:
hidden:
_compute_wind_rose()
_run_wind_rose()
Public:
compute()
run()

File: rex/utilities/utilities.py (hunk 1 above)
Comment (MRossol) [reply rex/utilities/utilities.py:610]: that might be outdated, i'll update

File: rex/renewable_resource.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (grantbuster) [reply rex/renewable_resource.py:714]: No if it's a property we would just load the attrs if requested. would be nice to be able to call resource.attrs to get a high level view of attributes. Not necessary though.

**Pull request merged**